### DISCUSSION TITLE
Testing add to cart behaviour

### DISCUSSION THREAD

**Discussion by ShokoPeak** [jef/streetmerchant#1441] [Help]
Hi guys, sorry if this is a question with a simple or obvious answer, but is there an easy way to test the add to cart behaviour for a specific store? I'm trying to figure out whether the bot is trying to add a GPU to my cart on the cclonline.com website and just not getting it quick enough or whether there's something in the configuration I need to change.

Thank you, and thanks for your hard work on the project!

Edit: I've done some further research and testing. I'm not sure if this is of use to anyone or would help, but i've found that injecting the JS code `document.getElementById('btnAddToBasket').click();` the browser will attempt to add the product to the cart on the CCL website. I'm not sure how much this is possible to do with streetmerchant, but i'll keep digging and post whatever I find. 

Sorry, i'm a lapsed webdev and new to Node!

**Comment by ShokoPeak** [comment]
I did find a workaround for anyone interested. I used the chrome extension Tampermonkey to inject some JS after the doc has loaded to automatically add any card (or other item!) to your cart. Code snippet below.

```
// ==UserScript==
// @name         Auto Add to Cart CCL
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://www.yourwebsiteaddress.com/product/*
// @grant        none
// @run-at document-end
// ==/UserScript==

(function() {

    document.getElementById('idForYourAddToCartButton').click();

})();
```

You just need to update 'idForYourAddToCartButton' to the correct one for whichever website you're trying to auto add to cart on and change 'https://www.yourwebsiteaddress.com/product/*' accordingly to match the format of the target website. No extra setup required, should run the moment the webpage has done loading.

Sorry for the wild ride of comments, i'm learning as I go here!

**Comment by absolutelyNoClue** [comment]
setting AUTO_ADD_TO_CART="true" on your config will do the same.

**Reply by ShokoPeak** [reply]
First thing I tried, doesn't seem to be supported for a few websites though. Got it working with the workaround above.

**Reply by Heeere-GittyGitty** [reply]
> 
> 
> setting AUTO_ADD_TO_CART="true" on your config will do the same.

No it doesn't, it just issues a GET command. This triggers an action in the "add" button, which can call a Post for example, and has session cookie and form variables filled in.
This is actually the only way it works most of the times..

Thanks @ShokoPeak

**Reply by Heeere-GittyGitty** [reply]
Another solution would be to run a script when item in stock, and then you could programmatically call Curl with the session cookie you have for that site. But there are form generated keys which would prevent this to work most of the times.

**Comment by 85256638** [comment]
Thanks for the suggestion. I will give it a go, have you tried this on Currys?

**Reply by 85256638** [reply]
Awesome, it works! Thanksa lot @ShokoPeak !
my code below:

// ==UserScript==
// @name         Auto Add to Cart CCL
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match        https://www.cclonline.com/category/401/PC-Components/CPU-Processors/AMD-Processors-and-APUs/
// @grant        none
// @run-at document-end
// ==/UserScript==

(function() {

    document.getElementById('ctl00_complexproductlist_rptProducts_productItem_2_btnAddToBasket_2').click();

})();

**Comment by 85256638** [comment]
@ShokoPeak , just tested on Currys.co.uk, this code also works! For any items in Currys.co.uk, you would need to change the `document.getElementById` into `getElementsByClassName` 

my code below:

`  document.getElementsByClassName('Classidofyouraddtobasketbutton')[0].click();`

**Reply by swagsourav** [reply]
I am very new to coding, i wanted to know , once i have ADD_TO_CART=yes, and this usescript created on tampermonkey, how does this script gets triggered ?  and where do i fill in the session and form cookie ?

Any help is greatly appreciated

**Reply by swagsourav** [reply]
@85256638 @ShokoPeak 

For example If this is the product i am after
https://www.currys.co.uk/gbuk/product-**10191506**-pdt.html

Should this snippet document.getElementsByClassName('Class**idofyouraddtobasketbutton**')[0].click();

be changed to 

document.getElementsByClassName('Class**10191506**)[0].click();

**Reply by 85256638** [reply]
Hi, the link you gave me returned 404, could you give me the full URL without **** please?

**Reply by swagsourav** [reply]
@85256638 this is the actual link https://www.currys.co.uk/gbuk/product-10191506-pdt.html

**Reply by swagsourav** [reply]
@85256638 https://www.currys.co.uk/gbuk/computing-accessories/components-upgrades/graphics-cards/msi-geforce-gtx-1660-ti-6-gb-ventus-xs-oc-graphics-card-10191506-pdt.html

**Reply by swagsourav** [reply]
@85256638 This is what i create in tampermonkey, is that correct ????

// ==UserScript==
// @name Auto Add to Cart CURRYS
// @namespace http://tampermonkey.net/
// @Version 0.1
// @description try to take over the world!
// @author Me
// @match https://www.currys.co.uk/gbuk/computing-accessories/components-upgrades/graphics-cards/msi-geforce-gtx-1660-ti-6-gb-ventus-xs-oc-graphics-card-10191506-pdt.html
// @grant none
// @run-at document-end
// ==/UserScript==

(function() {

document.getElementsByClassName('Class10191506')[0].click();

})();

**Reply by 85256638** [reply]
// ==UserScript==
// @name         Auto Add Graphics Card at Currys.co.uk
// @namespace    http://tampermonkey.net/
// @version      0.1
// @description  try to take over the world!
// @author       You
// @match       https://www.currys.co.uk/gbuk/computing-accessories/components-upgrades/graphics-cards/*
// @grant        none
// @run-at document-end
// ==/UserScript==

(function() {

    document.getElementsByClassName('Button__StyledButton-bvTPUF hZIOeU Button-jyKNMA GZkwS')[0].click();

})();


Try this one please, works for any graphics card on Currys.co.uk, let me know if this works for you.

**Reply by swagsourav** [reply]
@85256638 This works perfectly fine, i just tested it on a graphics card which was showing in stock and it added it to my cart. Thank you so much for your help. Much appreciated

**Reply by 85256638** [reply]
@swagsourav Glad to hear, this is would be the upper limit of my skill, anything beyond after adding to basket, I would love to see someone could help out... Can't really beat the bot at all, even the card was added to the basket, when I tried to click pay by card, it will stuck at "can't process this order"...

**Reply by swagsourav** [reply]
@85256638 i am frustrated too, i think i have tried 6 odd times all within a min and still loose out. I have my processor , motherboard and Ram, already bought. Honestly if i would have know this is what i am up against , i would have done my build next year

**Reply by 85256638** [reply]
exactly, and now we have to face the challenge from the cryptocurrency miners.
I would suggest you buy AMD cards, easier to buy than the Nvdia's
